### PR TITLE
Add truncate_str_short() which always returns a prefix of the input

### PR DESCRIPTION
**Pull request by th1000s**
Document that the input of `truncate_str()` may be cut on a double width char ("fullwidth") - then the output is no longer a prefix of the input.

`truncate_str_short()` could be faster by not constructing the result but only returning an index, but it will be used in rare edge cases so reusing existing logic is good enough.

Common code moved to `truncate_str_impl()`.

Assume (with fallback) that graphemes are at most 2 wide.
Fewer allocations.
Tests.

### CODE REVIEW

File: src/ansi/mod.rs [hunk 1]
```diff
@@ -51,20 +46,60 @@ pub fn truncate_str<'a>(s: &'a str, display_width: usize, tail: &str) -> Cow<'a,
...
+pub fn truncate_str<'a>(s: &'a str, display_width: usize, tail: &str) -> Cow<'a, str> {
+    truncate_str_impl(s, display_width, tail, Some(' '))
+}
+
+/// Truncate string `s` so it fits into `display_width`, ignoring any ANSI escape sequences when
+/// calculating the width. If a double-width ("fullwidth") grapheme has to be cut, it is omitted and
+/// the resulting string is *shorter* than `display_width`. But this way the result is always a
+/// prefix of the input `s`.
+pub fn truncate_str_short(s: &str, display_width: usize) -> Cow<str> {
+    truncate_str_impl(s, display_width, "", None)
```
Comment (dandavison) [inline src/ansi/mod.rs:102]: clippy wants `#[allow(dead_code)]` here

Comment (th1000s) [reply src/ansi/mod.rs:102]: I'll just merge it with the  --max-syntax-highlighting-length PR :)

**Comment by th1000s** [comment]
Now contained in PR #1746
